src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeLoaderEnvironment.cs

```diff
@@ -109,7 +110,32 @@ public sealed partial class TypeLoaderEnvironment
         [ThreadStatic]
         private static bool t_isReentrant;
 
-        public static TypeLoaderEnvironment Instance { get; } = new TypeLoaderEnvironment();
+        private static TypeLoaderEnvironment s_instance;
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static TypeLoaderEnvironment InitializeInstance()
+        {
+            TypeLoaderEnvironment instance = new TypeLoaderEnvironment();
+            if (Interlocked.CompareExchange(ref s_instance, instance, null) == null)
+                return instance;
+            return s_instance;
+        }
+
+        public static TypeLoaderEnvironment Instance
+        {
+            get
+            {
+                return s_instance ?? InitializeInstance();
+            }
+        }
+
+        public static TypeLoaderEnvironment InstanceOrNull
```

Comment:
Any reason not to use expression bodied form

Reply:
I'll double check when I'm at a computer but the type loader codebase is legacy and doesn't use them